sw/device/lib/crypto/include/api.h

```diff
@@ -700,29 +696,85 @@ crypto_status_t otcrypto_aes_decrypt_gcm(const crypto_blinded_key_t *key,
                                          crypto_uint8_buf_t *plaintext);
 
 /**
- * Performs the internal GHASH operation of Galois Counter Mode (GCM).
- *
- * This function is an operation internal to GCM and can be used to
- * create custom implementations that do not adhere to the AES-GCM
- * encryption and decryption APIs provided here. However, custom GCM
- * constructs can be dangerous; for most use cases, prefer the
- * provided encryption and decryption operations.
+ * Internal GHASH operation of Galois Counter Mode (GCM).
  *
- * The length of the input bitstring must be a multiple of 128 bits,
- * and must not be zero.
+ * GHASH is an operation internal to GCM. It can be used to create
+ * custom implementations that do not adhere to the AES-GCM encryption
+ * and decryption API provided here. However, custom GCM constructs
+ * can be dangerous; for most use cases, prefer the provided
+ * encryption and decryption operations.
  *
- * The caller should allocate space for the `output` buffer, which
- * must have a length of 16 bytes.
+ * This function initializes the GHASH context and must be called
+ * to create a context object.
  *
  * @param hash_subkey Hash subkey (H), 16 bytes
- * @param input Input bitstring (X)
- * @param output buffer, 16 bytes
- * @return crypto_status_t Result of the authenticated decryption
- * operation
+ * @param ctx Output GHASH context object, caller-allocated
+ * @return crypto_status_t Result of the operation
```

Comment:
(applies to other doc comments as well)

```suggestion
 * @return Result of the operation
```